### PR TITLE
[MM-28323] Don't show old tutorial or what's new modal for cloud instances

### PR DESCRIPTION
**Pull request by nickmisasi**
### Summary
On first login, users should be brought directly to the next steps view, skipping the what's new modal, and the old style tutorial.

Test steps:
Invite yourself to the spinwick server
login
you should be directed right to the next steps view 

(requires cloud license)

Remove the cloud license
Do the above
You should see the what's new and old style tutorial

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28323
#### Related Pull Requests
n/a

#### Screenshots
n/a

### CODE REVIEW

File: components/sidebar/sidebar.tsx [hunk 1]
```diff
@@ -58,11 +59,11 @@ export default class Sidebar extends React.PureComponent<Props, State> {
             this.props.actions.fetchMyCategories(this.props.teamId);
         }
 
-        if (!this.props.hasSeenModal) {
-            this.props.actions.openModal({
-                modalId: ModalIdentifiers.SIDEBAR_WHATS_NEW_MODAL,
-                dialogType: SidebarWhatsNewModal,
-            });
+        if (!this.props.hasSeenModal && !this.props.isCloud) {
+          this.props.actions.openModal({
```
Comment (devinbinnie) [inline components/sidebar/sidebar.tsx:63]: Nit: indenting

**Review by devinbinnie — APPROVED** [review]
One non-blocking comment

**Comment by nickmisasi** [comment]
Forgot that Cloud is string `'true'` not boolean `true` @devinbinnie changed the PR if you want to take another look!

**Review by marianunez — APPROVED** [review]
LGTM, just a thought below

File: components/sidebar/sidebar.tsx [hunk 2]
```diff
@@ -58,7 +59,7 @@ export default class Sidebar extends React.PureComponent<Props, State> {
             this.props.actions.fetchMyCategories(this.props.teamId);
         }
 
-        if (!this.props.hasSeenModal) {
+        if (!this.props.hasSeenModal && !this.props.isCloud) {
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply components/sidebar/sidebar.tsx:62]: It wasn't, is there a way of detecting that? If so we should create a bug to fix that, cause I think that would be ideal.

**Review by stevemudie — APPROVED** [review]
Worked as expected.

**Pull request merged**